src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -60,4 +70,15 @@
       });
   }
 
+  @Validate
+  @Override
+  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
+                         Handler<AsyncResult<Response>> handler, Context context) {
+    // delete Kafka topics if tenant purged
+    Future<Void> result = tenantAttributes.getPurge() != null && tenantAttributes.getPurge()
+      ? new KafkaAdminClientService(context.owner()).deleteKafkaTopics(TenantTool.tenantId(headers), environmentName())
+      : Future.succeededFuture();
+    result.onComplete(x -> super.postTenant(tenantAttributes, headers, handler, context));
```

Comment:
same here regarding copying the existing implementation from mod-inventory-storage...
even if the topics are not deleted and tenant is being recreating, this shouldn't be a problem because the topic creation procedure creates ONLY the topics that don't exist